joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -161,10 +204,16 @@ def configure_robot(self, description):
         self.free_joints = {}
         self.joint_list = [] # for maintaining the original order of the joints
 
-        if robot.getElementsByTagName('COLLADA'):
+        # Get root tag to parse file format
+        root = robot.documentElement
+        if root.tagName == 'sdf':
+            self.init_sdf(robot)
+        elif root.tagName == 'COLLADA':
             self.init_collada(robot)
-        else:
+        elif root.tagName == 'urdf':
```

Comment:
I don't think that this is correct; as far as I know, `urdf` files start with the `robot` tag: https://github.com/ros/urdfdom/blob/b5fffec979cabb9563f6f0ad3aebd92bdc978bae/xsd/urdf.xsd#L300

Reply:
Should I go for checking both urdf and robot tags or no checks directly parsing it as urdf?